langchain/document_loaders/roam.py

```diff
@@ -1,25 +1,27 @@
 """Loader that loads Roam directory dump."""
-from pathlib import Path
-from typing import List
+from typing import List, Generator
 
 from langchain.docstore.document import Document
 from langchain.document_loaders.base import BaseLoader
+from langchain.document_loaders.blob_loaders.file_system import FileSystemLoader
 
 
 class RoamLoader(BaseLoader):
     """Loader that loads Roam files from disk."""
 
-    def __init__(self, path: str):
+    def __init__(self, path: str) -> None:
         """Initialize with path."""
-        self.file_path = path
+        self.loader = FileSystemLoader(path, glob="**/*.md")
```

Comment:
what're your thoughts on having BlobLoaders as instance vars versus inheriting from them? e.g. an alternative implementation could be
```python
class RoamLoader(BaseLoader, FileSystemLoader):
  def __init__(self, path):
    super().__init__(path, glob="**/*.md")
    
  def lazy_load(self) -> ...:
    for blob in self.yield_blobs():
      yield ...
```

i don't have a good way of deciding b/n the two approaches, curious if you do

Reply:
I would prefer encouraging a compositional pattern, so the loader doesn't end up getting tied to a particular storage system (e.g., the file system).

i.e., if the `md` files are stored on s3 rather than the local file system, we'd want to swap out the loader instead of implementing another RoamLoader class that's specialized for s3